app/models/govuk_publishing_components/component_doc.rb

```diff
@@ -41,5 +43,26 @@ def partial_path
     def govspeak_to_html(govspeak)
       Govspeak::Document.new(govspeak).to_html
     end
+
+    def check_for_universal_accessibility_criteria
+      fetch_component_docs.map { |criteria| match_universal_accessibility_criteria(criteria) }.join("\n") if universal_accessibility_criteria.present?
+    end
+
+    def match_universal_accessibility_criteria(criteria)
+      criteria["accessibility_criteria"] if universal_accessibility_criteria.include? criteria["id"]
+    end
+
+    def fetch_component_docs
```

Comment:
I think we're mixing up concerns a little by having the ComponentDoc be responsible for fetching the shared criteria.

Reply:
Seems like something that should be a different class.

Reply:
It's also a bit copy-and-paste, these method names are not clear – eg not clearly different to those in the component resolver class.